core/src/main/scala/uk/gov/nationalarchives/tdr/api/core/graphql/GraphQlTypes.scala

```diff
@@ -104,6 +125,73 @@ object GraphQlTypes {
         "files",
         ListType(FileType),
         resolve = context => DeferConsignmentFiles(context.value.id)
+      ),
+      Field(
+        "offsetConnections",
+        filesConnections,
+        arguments = List(LimitArg, PageNumberArg),
+        resolve = ctx => {
+          val limit: Int = ctx.args.arg("limit")
+          val currentPage: Int = ctx.args.arg("pageNumber")
+          val offset: Int = (currentPage - 1) * limit
+          DeferredValue(DeferOffsetPagination(
+            ctx.value.id,
+            limit,
+            offset
+          )).map(
+            response => {
+              val edges = response._2
+              val totalCount = response._1
+              val totalPages = math.ceil(totalCount.toDouble / limit.toDouble)
+              val firstEdge = edges.headOption
+              val lastEdge = edges.lastOption
+
+              DefaultConnection(
+                PageInfo(
+                  startCursor = firstEdge.map(_.cursor),
+                  endCursor = lastEdge.map(_.cursor),
+                  hasNextPage = currentPage < totalPages,
+                  hasPreviousPage = 1 < currentPage
+                ),
+                edges
+              )
+            }
+          )
+        }
+      ),
+      Field(
+        "keySetConnections",
+        filesConnections,
+        arguments = List(LimitArg, AfterArg, PageNumberArg),
```

Comment:
As we talked about in person, it would be nice if we didn't have to have a cursor _and_ a page numbers, since it ought to be possible to do pagination with just a cursor.

Reply:
Resolved